src/sqs/sqs.ts

```diff
@@ -299,7 +300,7 @@ export class SqsQueue<MessagePayload> {
 			if (isNotAFailure(errorHandlingResult)) {
 				context.log.info("Deleting the message because the error is not a failure");
 				await this.deleteMessage(context);
-			} else if (isNotRetryable(errorHandlingResult)) {
+			} else if (isNotRetryable(errorHandlingResult) || UNRETRYABLE_STATUS_CODES.includes(err.status)) {
```

Comment:
I see `src/sqs/error-handlers.ts` func `maybeHandleNonRetryableResponseCode` does a similar job. Can you please explain why isn't that sufficient but we have to do it here?